### ISSUE TITLE
Warnings in iOS, macOS builds in Release configuration

### ISSUE THREAD

**Issue by jfirebaugh** [mapbox/mapbox-gl-native#8314]
Running `make xpackage` produces the following warnings:

```
⚠️  /Users/john/Development/mapbox-gl-native/platform/macos/src/MGLMapView.mm:2430:26: unused variable 'overlay' [-Wunused-variable]
    for (id <MGLOverlay> overlay in overlays) {
                         ^

⚠️  /Users/john/Development/mapbox-gl-native/platform/macos/src/MGLMapView.mm:2441:26: unused variable 'overlay' [-Wunused-variable]
    for (id <MGLOverlay> overlay in overlays) {
                         ^

⚠️  /Users/john/Development/mapbox-gl-native/platform/macos/src/MGLMapView.mm:148:17: method definition for 'visibleFeaturesAtPoint:inStyleLayersWithIdentifiers:' not found [-Wincomplete-implementation]
@implementation MGLMapView {
                ^

⚠️  /Users/john/Development/mapbox-gl-native/platform/macos/src/MGLMapView.mm:148:17: method definition for 'visibleFeaturesInRect:inStyleLayersWithIdentifiers:' not found [-Wincomplete-implementation]
@implementation MGLMapView {
^

⚠️  /Users/john/Development/mapbox-gl-native/platform/darwin/src/MGLStyle.mm:344:21: unused variable 'customLayer' [-Wunused-variable]
    } else if (auto customLayer = mbglLayer->as<mbgl::style::CustomLayer>()) {
^
```

**Comment by 1ec5** [comment]
These warnings occur in Release configuration but not Debug configuration. We should make the two have consistent build settings for warnings (also on iOS).

**Comment by 1ec5** [comment]
The warnings about `-visibleFeaturesAtPoint:inStyleLayersWithIdentifiers:` were fixed in #8305 on the release-ios-v3.5.0-android-v5.0.0 branch.

**Comment by 1ec5** [comment]
Fixed in #8368 on the release-ios-v3.5.0-android-v5.0.0 branch.
